processing/src/main/java/io/druid/segment/ColumnSelectorBitmapIndexSelector.java

```diff
@@ -110,6 +110,21 @@ public void inspectRuntimeShape(RuntimeShapeInspector inspector)
   }
 
   @Override
+  public boolean hasMultipleValues(final String dimension)
+  {
+    if (isFilterableVirtualColumn(dimension)) {
```

Comment:
According to how the method `hasMultipleValues()` is called and no javadocs, the implementation shouldn't check that the column is filterable or not. This check should be moved to `ExpressionFilter`.